sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/BlobUploadEntity.java

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob.transfer;
+
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.Ignore;
+import androidx.room.PrimaryKey;
+import androidx.room.TypeConverters;
+
+import java.io.File;
+import java.util.Objects;
+
+/**
+ * Package private.
+ *
+ * Represents blob upload metadata for a single file upload.
+ *
+ * The Data Access Object type {@link UploadDao} exposes DB store and read methods on this model.
+ *
+ * @see TransferDatabase
+ */
+@Entity(tableName = "blobuploads")
+final class BlobUploadEntity {
```

Comment:
We'd discussed in arch meeting about maybe making this generic and having "source" and "destination" properties. Would this PR be the best place to do that or did you want to address later?

Reply:
(This comment would mostly apply to anywhere else that something is titled "Upload...")

Reply:
right, I'm thinking of considering it once we have full download APIs in place as well, so not planning in this PR.

Reply:
you might have already noticed, all types (including Upload prefixed types) package internal, the only public type is `TransferClient`.

Reply:
you might have already noticed, all types (including Upload prefixed types) are package internal, the only public type is `TransferClient`.